weedcoco/importers/labelbox.py

```diff
@@ -0,0 +1,194 @@
+        labelSet = labelboxJSON[i]
+        try:
+            objectList = labelSet['Label']['objects']
+        except KeyError:
+            print('[ERROR] {} skipped - not labelled'.format(labelSet['External ID']))
+
+        if check_labelbox_review_score(labelSet) == 1 and len(objectList) > 0:
+            try:
+                sequentialImageName = "image_" + str(i) + ".jpg"
+                # remove any extra spaces in the name
+                imageLog[sequentialImageName] = labelSet['External ID'].replace(" ", "")
+
+                # keep the original image names on labelbox
+                if keepExternalID:
+                    imageName = labelSet['External ID'].replace(" ", "")
+                    vocName = labelSet['External ID'].split('.')[0] + '.xml'
+                else:
+                    imageName = sequentialImageName
+                    vocName = sequentialImageName.split('.')[0] + ".xml"
+
+                # get the image URL and download the data
+                origImgURL = labelSet['Labeled Data']
+                image, imagePath = download_png(origImgURL, imageName, saveImageDir)
+
+                # instantiate the VOC writer
+                vocWriter = Writer(imagePath, height=image.shape[0], width=image.shape[1])
+
+                # iterate over each object in the list
+                for object in objectList:
+                    className = object['title']
+                    bbox = object['bbox']
+                    startX = bbox['left']
+                    startY = bbox['top']
+                    endX = startX + bbox['width']
+                    endY = startY + bbox['height']
+
+                    vocWriter.addObject(className, startX, startY, endX, endY)
+                vocPath = os.path.join(saveVOCDir, vocName)
+                vocWriter.save(vocPath)
+
+            except KeyError:
+                print("[INFO] No objects found")
+        else:
+            print("[INFO] Image {} skipped".format(i))
+
+    return Path(saveImageDir), Path(saveVOCDir)
+
+def main():
+    ap = argparse.ArgumentParser(description=__doc__)
+
+    ap.add_argument("--labelbox-json", required=True, type=str, help='path to downloaded labelbox JSON file')
+    ap.add_argument("--save-dir", required=True, type=str)
+    ap.add_argument("--keep-original-ID", action="store_true", default=False)
+    ap.add_argument("--voc-to-coco", action="store_true", default=False)
+    ap.add_argument("--start-index", default=0, type=int)
+    ap.add_argument(
+        "--category-name-map",
+        type=Path,
+        help="JSON or YAML mapping of VOC names to WeedCOCO category names")
+    ap.add_argument("--metadata-path", type=Path)
+    ap.add_argument("--agcontext-path", type=Path)
+    ap.add_argument("--save-image-dir", type=str, help='existing save directory for images')
+    ap.add_argument("--save-voc-dir", type=str, help='existing save directory for VOC files')
```

Comment:
I'm not convinced we need the functionality of saving the VOC. Should be saved to a temp directory, then converted, in the present context of weedcoco?

Reply:
It had been a bit of a hack workaround because both were available to me and it was simple to join one to the other, but yes just going straight to weedCOCO would be best maybe with the option to save VOC along the way? Or is that YAGNI